core/src/main/java/org/hisp/dhis/android/core/indicator/IndicatorTypeFields.java

```diff
@@ -28,21 +28,28 @@
 
 package org.hisp.dhis.android.core.indicator;
 
-import org.hisp.dhis.android.core.common.ModelBuilder;
-
-public class IndicatorTypeModelBuilder extends ModelBuilder<IndicatorType, IndicatorTypeModel> {
-
-    @Override
-    public IndicatorTypeModel buildModel(IndicatorType type) {
-        return IndicatorTypeModel.builder()
-                .uid(type.uid())
-                .code(type.code())
-                .name(type.name())
-                .displayName(type.displayName())
-                .created(type.created())
-                .lastUpdated(type.lastUpdated())
-                .number(type.number())
-                .factor(type.factor())
-                .build();
+import org.hisp.dhis.android.core.arch.fields.FieldsHelper;
+import org.hisp.dhis.android.core.data.api.Field;
+import org.hisp.dhis.android.core.data.api.Fields;
+
+public final class IndicatorTypeFields {
+
+    final static String NUMBER = "number";
+    final static String FACTOR = "factor";
+
+    private static final FieldsHelper<IndicatorType> fh = new FieldsHelper<>();
+
+    public static final Field<IndicatorType, String> uid = fh.uid();
+
+    static final Field<IndicatorType, String> lastUpdated = fh.lastUpdated();
+
+    public static final Fields<IndicatorType> allFields = Fields.<IndicatorType>builder()
+            .fields(fh.getIdentifiableFields())
```

Comment:
@marcamsn IndicatorType extends BaseNameable

Reply:
@vgarciabnz The IndicatorType has been left extending IdentifiableObjects and the migration has been done to eliminate the unnecessary columns in the database.